packages/pigeon/tool/shared/generation.dart

```diff
@@ -118,7 +118,7 @@ Future<int> generateTestPigeons(
           ? null
           : '$outputBase/ios/Classes/$pascalCaseName.gen.swift',
       swiftErrorClassName: swiftErrorClassName,
-      swiftIncludeErrorClass: input != 'core_tests',
+      swiftIncludeErrorClass: input != 'background_platform_channels',
```

Comment:
Fixes a test failure caused by https://github.com/flutter/packages/pull/7849

`core_tests` and `background_platform_channels` both generated the `PigeonError` class which causes conflicts. 

**tldr:**
This wasn't caught by CI since `core_tests` is a checked in and the order of the tests matters.

When I reran the generator and pushed my code, it removed the `PigeonError` from `core_tests`. Therefore the tests that run before the CI generates the unchecked in files would fail because `core_tests` was missing the reference to `PigeonError` that was in `background_platform_channels`. Therefore the fix was to switch the location the single `PigeonError` to reside in `core_tests` instead of `background_platform_channels`.

Reply:
This is also addressed in my current pr. Your solution is smarter than mine :)